src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -618,17 +620,51 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
                 }).ConfigureAwait(false);
             }
 
-                foreach (var loadOption in _options.KeyValueSelectors)
+            foreach (KeyValueSelector loadOption in _options.KeyValueSelectors)
+            {
+                IAsyncEnumerable<ConfigurationSetting> settingsEnumerable;
+
+                if (string.IsNullOrEmpty(loadOption.SnapshotName))
                 {
-                    var selector = new SettingSelector
+                    settingsEnumerable = client.GetConfigurationSettingsAsync(
+                        new SettingSelector
+                        {
+                            KeyFilter = loadOption.KeyFilter,
+                            LabelFilter = loadOption.LabelFilter
+                        },
+                        cancellationToken);
+                }
+                else
+                {
+                    ConfigurationSettingsSnapshot snapshot;
+
+                    try
+                    {
+                        snapshot = await client.GetSnapshotAsync(loadOption.SnapshotName).ConfigureAwait(false);
+                    }
+                    catch (RequestFailedException rfe)
                     {
-                        KeyFilter = loadOption.KeyFilter,
-                        LabelFilter = loadOption.LabelFilter
-                    };
+                        if (rfe.Status == (int)HttpStatusCode.NotFound)
+                        {
+                            throw new InvalidOperationException($"Could not load snapshot with name '{loadOption.SnapshotName}'");
```

Comment:
We should be specific. The snapshot was not found. Saying we could not load it could allude to a multitude of issues.